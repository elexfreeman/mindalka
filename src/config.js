export const BREAKPOINTS = Object.freeze({
  sm: 1020,
  md: 1440,
  lg: 1660,
});

export const API_URL = `http://lavacacao.ru:1337/api`;
export const MEDIA_URL = `http://lavacacao.ru:1337`;
