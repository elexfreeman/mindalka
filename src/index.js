
export const parser = (data) => {
  return data;
};
