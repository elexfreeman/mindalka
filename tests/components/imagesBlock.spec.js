/* eslint-disable i18n/no-russian-character, max-len */
import { parser } from '../../src/index';

describe('Images block ', () => {
  it('Check pattern full', () => {
    const src = `
<!-- component:imagesBlock
  smRow: 3
  mdRow: 3
  lgRow: 3
  data
-->
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
###### title *1*
description 2
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
description 3
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
#### title *4*
description 4
<!-- /component -->
`;

    expect(parser(src)).toMatchSnapshot();
  });

  it('Check pattern img only', () => {
    const src = `
<!-- component:imagesBlock
  smRow: 3
  mdRow: 3
  lgRow: 3
  data
-->
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
<!-- /component -->
`;

    expect(parser(src)).toMatchSnapshot();
  });

  it('Check pattern img title', () => {
    const src = `
<!-- component:imagesBlock
  smRow: 3
  mdRow: 3
  lgRow: 3
-->
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
### title 111

![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
## title 222
<!-- /component -->
`;

    expect(parser(src)).toMatchSnapshot();
  });

  it('Check pattern img description', () => {
    const src = `
<!-- component:imagesBlock
  smRow: 3
  mdRow: 3
  lgRow: 3
-->
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
description 1

![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
description 2
<!-- /component -->
`;

    expect(parser(src)).toMatchSnapshot();
  });

  it('Check mix', () => {
    const src = `
<!-- component:imagesBlock
  smRow: 3
  mdRow: 3
  lgRow: 3
-->
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
###### title *1*
description 2
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
description 3
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
<!-- /component -->
`;

    expect(parser(src)).toMatchSnapshot();
  });

  it('Check wrong component', () => {
    const src = `
<!-- component:imagesBlock
  smRow: 3
  mdRow: 3
  lgRow: 3
-->
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
# title *1*
description 2
description 2

![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
description 3
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
![картинка center](https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg)
<!-- /component -->
`;

    expect(parser(src)).toMatchSnapshot();
  });
});

/* eslint-enable i18n/no-russian-character, max-len */
