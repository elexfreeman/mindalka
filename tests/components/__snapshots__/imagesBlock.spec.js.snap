// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Images block  Check mix 1`] = `
"

<div class=\\"mdc-images-block__smrow-3 mdc-images-block__mdrow-3 mdc-images-block__lgrow-3 mdc-images-block\\">

<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  <h6>title <em>1</em></h6>

  <p>description 2</p>

</div>
<!-- end mdc-image --> 
    
<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  <p>description 3</p>

</div>
<!-- end mdc-image --> 
    
<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  
</div>
<!-- end mdc-image --> 
    
<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  
</div>
<!-- end mdc-image --> 
    
</div>
"
`;

exports[`Images block  Check pattern full 1`] = `
"

<div class=\\"mdc-images-block__smrow-3 mdc-images-block__mdrow-3 mdc-images-block__lgrow-3 mdc-images-block__data-undefined mdc-images-block\\">

<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  <h6>title <em>1</em></h6>

  <p>description 2</p>

</div>
<!-- end mdc-image --> 
    
<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  <p>description 3</p>

</div>
<!-- end mdc-image --> 
    
<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  
</div>
<!-- end mdc-image --> 
    
<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  <h4>title <em>4</em></h4>

  <p>description 4</p>

</div>
<!-- end mdc-image --> 
    
</div>
"
`;

exports[`Images block  Check pattern img description 1`] = `
"

<div class=\\"mdc-images-block__smrow-3 mdc-images-block__mdrow-3 mdc-images-block__lgrow-3 mdc-images-block\\">

<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  <p>description 1</p>

</div>
<!-- end mdc-image --> 
    

<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  <p>description 2</p>

</div>
<!-- end mdc-image --> 
    
</div>
"
`;

exports[`Images block  Check pattern img only 1`] = `
"

<div class=\\"mdc-images-block__smrow-3 mdc-images-block__mdrow-3 mdc-images-block__lgrow-3 mdc-images-block__data-undefined mdc-images-block\\">

<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  
</div>
<!-- end mdc-image --> 
    
<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  
</div>
<!-- end mdc-image --> 
    
<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  
</div>
<!-- end mdc-image --> 
    
</div>
"
`;

exports[`Images block  Check pattern img title 1`] = `
"

<div class=\\"mdc-images-block__smrow-3 mdc-images-block__mdrow-3 mdc-images-block__lgrow-3 mdc-images-block\\">

<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  <h3>title 111</h3>

  
</div>
<!-- end mdc-image --> 
    

<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  <h2>title 222</h2>

  
</div>
<!-- end mdc-image --> 
    
</div>
"
`;

exports[`Images block  Check wrong component 1`] = `
"

<div class=\\"mdc-images-block__smrow-3 mdc-images-block__mdrow-3 mdc-images-block__lgrow-3 mdc-images-block\\">

<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  
</div>
<!-- end mdc-image --> 
    # title <em>1</em>
description 2
description 2


<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  <p>description 3</p>

</div>
<!-- end mdc-image --> 
    
<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  
</div>
<!-- end mdc-image --> 
    
<!-- mdc-image -->
<div class=\\"mdc-image-block\\">
  <img alt=\\"картинка center\\" src=\\"https://www.letu.ru/common/img/banners/gomobile_apr22_3320.jpg\\" />
  
  
</div>
<!-- end mdc-image --> 
    
</div>
"
`;
